File: backend/routes/price_alert.py
from fastapi import APIRouter

from models.price_alert import PriceAlert

from services.price_alert_service import (
    create_alert,
    get_alerts,
    delete_alert,
    check_price_alerts
)

router = APIRouter(
    prefix="/price-alert",
    tags=["Price Alerts"]
)


@router.post("/create")
def create(item: PriceAlert):

    return create_alert(item)


@router.get("/{email}")
def get(email: str):

    return get_alerts(email)
@router.delete("/delete")
def delete(
    email: str,
    symbol: str,
    target_price: float
):

    return delete_alert(
        email,
        symbol,
        target_price
    )


@router.get("/check/all")
def check():

    return check_price_alerts()

File: frontend/src/services/priceAlert.js
import api from "./api";

export const createAlert = async (
  email,
  symbol,
  target_price,
  condition
) => {

  const response = await api.post(
    "/price-alert/create",
    {
      email,
      symbol,
      target_price,
      condition,
    }
  );

  return response.data;
};

export const getAlerts = async (email) => {

  const response = await api.get(
    `/price-alert/${email}`
  );

  return response.data;
};

export const deleteAlert = async (
  email,
  symbol,
  target_price
) => {

  const response = await api.delete(
    "/price-alert/delete",
    {
      params: {
        email,
        symbol,
        target_price,
      },
    }
  );

  return response.data;
};